### PR TITLE
Fixed array_tests for Spark 4.0.0 [databricks]

### LINKED ISSUE #11007: Fix tests failures in array_test.py
These are the failing tests in array_test.py
```
FAILED ../../../../integration_tests/src/main/python/array_test.py::test_array_contains
FAILED ../../../../integration_tests/src/main/python/array_test.py::test_array_element_at
FAILED ../../../../integration_tests/src/main/python/array_test.py::test_array_element_at_ansi_fail_invalid_index
FAILED ../../../../integration_tests/src/main/python/array_test.py::test_array_element_at_zero_index_fail
FAILED ../../../../integration_tests/src/main/python/array_test.py::test_array_item
FAILED ../../../../integration_tests/src/main/python/array_test.py::test_array_item_ansi_fail_invalid_index
FAILED ../../../../integration_tests/src/main/python/array_test.py::test_array_item_lit_ordinal
FAILED ../../../../integration_tests/src/main/python/array_test.py::test_array_transform
FAILED ../../../../integration_tests/src/main/python/array_test.py::test_make_array
```

### PR DESCRIPTION
**Pull request by razajafri**
This PR fixes array_test by setting the `spark.sql.ansi.enabled` to false and some exception messages were changed.

fixes #11007

### CODE REVIEW

**Review by jlowe — COMMENTED** [review]
The description mentions that the ANSI config was set to false to fix some of these tests, but I don't see that change?

**Comment by razajafri** [comment]
build

**Comment by razajafri** [comment]
@jlowe I was waiting on #11006 to be merged. Can you PTAL on this

File: integration_tests/src/main/python/array_test.py [hunk 1]
```diff
@@ -103,11 +103,13 @@
 
 @pytest.mark.parametrize('data_gen', array_item_test_gens, ids=idfn)
 @pytest.mark.parametrize('index_gen', array_index_gens, ids=idfn)
+@disable_ansi_mode
```
Comment (jlowe) [inline integration_tests/src/main/python/array_test.py:106]: Is there a followup issue to remove this marker and fix this properly by explicitly setting the ANSI modes for tests?  This should not be a long-term fix for ANSI-related testing.

Comment (razajafri) [reply integration_tests/src/main/python/array_test.py:106]: [Yes](https://github.com/NVIDIA/spark-rapids/issues/11106), the plan is to revisit the tests and re-write tests and possibly incorporate the changes we discussed offline

**Review by jlowe — APPROVED** [review]

**Pull request merged**